src/libpgagroal/prometheus.c

```diff
@@ -1110,53 +1119,76 @@ metrics_page(int client_fd)
    char time_buf[32];
    int status;
    struct message msg;
+   struct prometheus* prometheus;
+
+   prometheus = (struct prometheus*)prometheus_shmem;
 
    memset(&msg, 0, sizeof(struct message));
 
-   now = time(NULL);
+   // can serve the message out of cache?
```

Comment:
If we are using the cache then we need to do locking with `prometheus_lock` such that we don't serve a partial result